packages/host/app/components/operator-mode/interact-submode.gts

```diff
@@ -487,74 +494,90 @@ export default class InteractSubmode extends Component<Signature> {
     return this.allStackItems.length > 0 && this.stacks.length === 1;
   }
 
-  @action private openSelectedSearchResultInStack(card: CardDef) {
-    let searchSheetTrigger = this.searchSheetTrigger; // Will be set by showSearchWithTrigger
-
-    // In case the left button was clicked, whatever is currently in stack with index 0 will be moved to stack with index 1,
-    // and the card will be added to stack with index 0. shiftStack executes this logic.
-    if (
-      searchSheetTrigger ===
-      SearchSheetTriggers.DropCardToLeftNeighborStackButton
-    ) {
-      for (
-        let stackIndex = this.stacks.length - 1;
-        stackIndex >= 0;
-        stackIndex--
-      ) {
-        this.operatorModeStateService.shiftStack(
-          this.stacks[stackIndex],
-          stackIndex + 1,
-        );
-      }
-
-      let stackItem: StackItem = {
-        card,
-        format: 'isolated',
-        stackIndex: 0,
-      };
-      this.operatorModeStateService.addItemToStack(stackItem);
-
-      // In case the right button was clicked, the card will be added to stack with index 1.
-    } else if (
-      searchSheetTrigger ===
-      SearchSheetTriggers.DropCardToRightNeighborStackButton
-    ) {
-      this.operatorModeStateService.addItemToStack({
-        card,
-        format: 'isolated',
-        stackIndex: this.stacks.length,
-      });
-    } else {
-      // In case, that the search was accessed directly without clicking right and left buttons,
-      // the rightmost stack will be REPLACED by the selection
-      let numberOfStacks = this.operatorModeStateService.numberOfStacks();
-      let stackIndex = numberOfStacks - 1;
-      let stack: Stack | undefined;
+  private openSelectedSearchResultInStack = restartableTask(
```

Comment:
the diff looks pretty heady, but the main this here is that async was introduced here, so this action turned into a task, which didn't diff very nicely